src/PhpPact/Consumer/Model/ConsumerRequest.php

```diff
@@ -88,28 +88,31 @@ public function getHeaders()
      */
     public function setHeaders(array $headers): self
     {
-        $this->headers = $headers;
+        $this->headers = [];
+        foreach ($headers as $header => $value) {
+            $this->addHeader($header, $value);
+        }
 
         return $this;
     }
 
     /**
-     * @param string       $header
-     * @param array|string $value
```

Comment:
`pactffi_with_header_v2` can support multiple values and matching rules. But I implement it incorrect. So I think it's better to move it to other pull request.


https://github.com/pact-foundation/pact-reference/blob/master/rust/pact_ffi/src/mock_server/handles.rs#L1038

Reply:
I added a commit to this PR anyway. See [bb256ef](https://github.com/pact-foundation/pact-php/pull/292/commits/bb256ef552c8d98860d601002a5382563130f237)